microsetta_interface/implementation.py

```diff
@@ -1158,13 +1277,33 @@ def get_fill_source_survey(*,
     if has_error:
         return survey_output
 
+    # We need to know if their consent is current if they're trying to take an
+    # external survey. Profiles without current consent can view their surveys
+    # but we'll block them from updating them on the POST action.
+    need_reconsent = check_current_consent(account_id, source_id, "data")
```

Comment:
Should this be handled by a `prerequisite`?

Reply:
...I'm not really sure from the flow so genuine question

Reply:
I didn't use prerequisites for the need_reconsent decision-making because in many of the cases, we're allowing them to access a given API path but with reduced functionality. There probably would be a way to use the prerequisite system, but I think it would end up more complex and prone to issues than the current code if/when it needs to be maintained.

Reply:
kk